### PR TITLE
[AIR] Update `WandbLoggerCallback` example

### LINKED ISSUE #31217: [AIR] Add docstring example of `WandbLoggerCallback`
### Description

Title

### Link

https://docs.ray.io/en/latest/ray-air/package-ref.html#weights-and-biases

### PR DESCRIPTION
**Pull request by bveeramani**
Signed-off-by: Balaji Veeramani <balaji@anyscale.com>

## Why are these changes needed?

The `WandbLoggerCallback` isn't tested, and it uses outdated APIs (`tune.run` instead of `Tuner.fit`). This PR updates the example so it's tested and uses recommended APIs.

## Related issue number

Closes #31217 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: .buildkite/pipeline.build.yml [hunk 1]
```diff
@@ -573,6 +573,7 @@
     - pip uninstall -y ray && rm -rf /ray/python/ray/thirdparty_files
     - pushd /ray && git clean -f -f -x -d -e .whl -e python/ray/dashboard/client && popd
     - bazel clean --expunge
+    - python ./ci/env/setup_credentials.py wandb
```
Comment (krfricke) [inline .buildkite/pipeline.build.yml:576]: Note that this will fail on PR builds. Can we somehow skip the test in that case?

**Review by krfricke — APPROVED** [review]
LGTM but one question for testing

File: .buildkite/pipeline.build.yml (hunk 1 above)
Comment (bveeramani) [reply .buildkite/pipeline.build.yml:576]: Could we add credentials to PR builds?

Once we make all code snippets runnable, we'll need credentials for other docstring examples.

Comment (krfricke) [reply .buildkite/pipeline.build.yml:576]: It's not straightforward as anyone could file a PR that exposes the secrets. This is why we have different permissions set-up for branch builds (e.g. master) and PRs - we do control what we merge, but we don't control the PRs anyone can open.

**Pull request merged**